src/tools/chunk_fs_hashing.js

```diff
@@ -0,0 +1,136 @@
+        const content_md5 = crypto.createHash('md5').update(data).digest('hex');
+        // Using async generator function in order to push data in small chunks
+        const source_stream = stream.Readable.from(async function*() {
+            for (let i = 0; i < data.length; i += CHUNK) {
+                yield data.slice(i, i + CHUNK);
+            }
+        }());
+        const target = new TargetHash();
+        const chunk_fs = new ChunkFS({
+            target_file: target,
+            fs_account_config: DEFAULT_FS_CONFIG,
+            rpc_client: DUMMY_RPC,
+            namespace_resource_id: 'MajesticSloth'
+        });
+        await stream_utils.pipeline([source_stream, chunk_fs]);
+        await stream_utils.wait_finished(chunk_fs);
+        const write_hash = target.digest();
+        console.log(
+            'Hash target',
+            `NativeMD5=${chunk_fs.digest}`,
+            `DataWriteCryptoMD5=${write_hash}`,
+            `DataOriginMD5=${content_md5}`,
+        );
+        assert.strictEqual(content_md5, write_hash);
+        assert.strictEqual(chunk_fs.digest, content_md5);
+        assert.strictEqual(chunk_fs.digest, write_hash);
+    });
+}
+
+async function file_target() {
+    fs.mkdirSync(F_PREFIX);
+    await P.map_with_concurrency(CONCURRENCY, Array(PARTS).fill(), async () => {
+        let target_file;
+        const data = crypto.randomBytes(PART_SIZE);
+        const content_md5 = crypto.createHash('md5').update(data).digest('hex');
+        const F_TARGET = path.join(F_PREFIX, content_md5);
+        try {
+            target_file = await nb_native().fs.open(DEFAULT_FS_CONFIG, F_TARGET, 'w', get_umasked_mode(config.BASE_MODE_FILE));
+            // Using async generator function in order to push data in small chunks
+            const source_stream = stream.Readable.from(async function*() {
+                for (let i = 0; i < data.length; i += CHUNK) {
+                    yield data.slice(i, i + CHUNK);
+                }
+            }());
+            const chunk_fs = new ChunkFS({
+                target_file,
+                fs_account_config: DEFAULT_FS_CONFIG,
+                rpc_client: DUMMY_RPC,
+                namespace_resource_id: 'MajesticSloth'
+            });
+            await stream_utils.pipeline([source_stream, chunk_fs]);
+            await stream_utils.wait_finished(chunk_fs);
+            const write_hash = crypto.createHash('md5').update(fs.readFileSync(F_TARGET)).digest('hex');
+            console.log(
+                'File target',
+                `NativeMD5=${chunk_fs.digest}`,
+                `DataWriteMD5=${write_hash}`,
+                `DataOriginMD5=${content_md5}`,
+            );
+            assert.strictEqual(content_md5, write_hash);
+            assert.strictEqual(chunk_fs.digest, content_md5);
+            assert.strictEqual(chunk_fs.digest, write_hash);
+            // Leave parts on error
+            fs.rmSync(F_TARGET);
+        } finally {
```

Comment:
For test tools I really think we can skip error handling like that - just letting errors throw will simply leave the open file handle for short moment until the tool exits anyhow.
Instead of `finally` I would just call close when we're done using the file (after wait_finished) - and no real need for `if(target_file)`.